install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,673 @@ toc::[]
 
 == Overview
 
-The HAProxy router is based on a
-link:http://golang.org/pkg/text/template/[*golang* template] that
-generates the HAProxy configuration file from a list of routes. If you
-want a customized template router to meet your needs, you can customize
-the template file, build a new container image, and run a customized router.
-Alternatively you can use a xref:../../dev_guide/configmaps.adoc#dev-guide-configmaps[ConfigMap].
-
-One common case for this might be implementing new features within the
-application back ends. For example, it might be desirable in a highly-available
-setup to xref:using-stick-tables[use stick-tables] that synchronizes between
-peers. The router plug-in provides all the facilities necessary to make this
-customization.
+The default HAProxy router is intended to satisfy the needs of most users.
+However, it does not expose all of the capability of HAProxy and users
+may need to modify the router for their own needs.
+
+You may need to implement new features within the application back-ends, or
+modify the current operation.  For example, it might
+be desirable in a highly-available setup to use
+xref:using-stick-tables[stick-tables] that synchronizes between peers. The
+router plug-in provides all the facilities necessary to make this customization.
+
+The router pod uses a template file to create the needed HAProxy configuration file.
+The template file is a
+link:http://golang.org/pkg/text/template/[golang template].
+When processing the template, the router has access to {product-title}
+information including the router's deployment configuration,
+the set of admitted routes and some helper functions.
+
+When the router pod starts and every time it reloads, it creates a HAProxy
+configuration file and then it starts HAProxy.  The
+link:https://cbonte.github.io/haproxy-dconv/configuration-1.5.html[Haproxy configuration manual]
+describes all of the features of HAProxy and how to construct a valid
+configuration file.
+
+A xref:using-configmap-replace-template[*configMap*] can be used to add the 
+new template to the router pod. With this approach, the router deployment
+configuration is modified to mount the *configMap* as a volume in the router
+pod.  The `TEMPLATE_FILE` environment variable is set to the full path name
+of the template file in the router pod.
+
+An alternative approach is to modify the *haproxy-template.config* file
```

Comment:
A second reason is to avoid the bother of setting up a config map making sure the desired set of router DCs mount it. Also, you don't need to run the same image in all the routers.